magda-esri-portal-connector/src/EsriPortal.ts

```diff
@@ -0,0 +1,563 @@
+
+    public getJsonDataset(id: string): Promise<any> {
+        const url = this.urlBuilder.getContentItemUrl(id);
+
+        return new Promise<any>((resolve, reject) => {
+            request(url, { json: true }, (error, response, body) => {
+                if (error) {
+                    reject(error);
+                    return;
+                }
+                resolve(body.result);
+            });
+        });
+    }
+
+    public getJsonDistributions(dataset: any): AsyncPage<any[]> {
+        return AsyncPage.single<object[]>(dataset.distributions || []);
+    }
+
+    public searchDatasetsByTitle(
+        title: string,
+        maxResults: number
+    ): AsyncPage<any[]> {
+        return undefined;
+    }
+
+    public getJsonFirstClassOrganizations(): AsyncPage<any[]> {
+        return undefined;
+    }
+
+    public getJsonFirstClassOrganization(id: string): Promise<any> {
+        return undefined;
+    }
+
+    public searchFirstClassOrganizationsByTitle(
+        title: string,
+        maxResults: number
+    ): AsyncPage<any[]> {
+        return undefined;
+    }
+
+    public getJsonDatasetPublisherId(dataset: any): string {
+        return undefined;
+    }
+
+    getJsonDatasetPublisher(dataset: any): Promise<any> {
+        return undefined;
+    }
+
+    private requestGroupInformation(contentId: string): Promise<any> {
+        return new Promise<any>((resolve, reject) => {
+            request(
+                this.urlBuilder.getContentItemGroups(contentId),
+                { json: true },
+                (error, response, body) => {
+                    if (error) {
+                        reject(error);
+                        return;
+                    }
+                    resolve(body);
+                }
+            );
+        });
+    }
+
+    private requestDistributionInformation(
+        distributionUrl: string
+    ): Promise<any> {
+        return new Promise<any>((resolve, reject) => {
+            request(
+                this.urlBuilder.getResource(distributionUrl),
+                { json: true },
+                (error, response, body) => {
+                    if (error) {
+                        reject(error);
+                        return;
+                    }
+                    resolve(body);
+                }
+            );
+        });
+    }
+
+    private requestDataSearchPage(
+        url: uri.URI,
+        startIndex: number
+    ): Promise<EsriPortalDataSearchResponse> {
+        const pageUrl = url.clone();
+        pageUrl.addSearch("start", startIndex);
+        pageUrl.addSearch("num", this.pageSize);
+        const that = this;
+        const operation = async () =>
+            new Promise<EsriPortalDataSearchResponse>((resolve, reject) => {
+                const requestUrl = pageUrl.toString();
+                console.log(
+                    `Requesting start = ${startIndex}, num = ${this.pageSize}`
+                );
+
+                request(
```

Comment:
Rather than another layer of callback, could this be an `await`? I think there's a promisified `request` library that we're using somewhere else, or we could use node-fetch?

Reply:
Have brought in node-fetch